src/charm.py

```diff
@@ -37,23 +38,31 @@ class VaultCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
-        self._stored.set_default(
-            root_token="", unseal_key="", root_ca="", root_token_provided=False
-        )
+        self._stored.set_default(root_token="", role_id="", unseal_key="")  # type: ignore[union-attr, arg-type]  # noqa: E501
```

Comment:
If we do drop the `totally_insecure` mode, we can drop `root_token` from here, and elsewhere in the code.